test/bigtest/tests/agreement-line-external-info-test.js

```diff
@@ -0,0 +1,232 @@
+/* eslint-disable camelcase */
+import React from 'react';
+import { describe, beforeEach, it } from '@bigtest/mocha';
+import chai from 'chai';
+import spies from 'chai-spies';
+import { StaticRouter as Router } from 'react-router-dom';
+import { mountWithContext } from '../helpers/mountWithContext';
+
+import { Info } from '../../../src/components/AgreementLineSections';
+
+import { externalEntitlements } from './resources';
+import AgreementLineInfoInteractor from '../interactors/agreement-line-external-info';
+
+chai.use(spies);
+const { expect } = chai;
+
+describe('AgreementLineInfo', () => {
+  const interactor = new AgreementLineInfoInteractor();
+
+  externalEntitlements.forEach((externalResource, i) => {
+    describe(`agreement line info of external resource of type '${externalResource?.reference_object?.type}' '${externalResource?.reference_object?.label}', index ${i}`, () => {
+      beforeEach(async () => {
+        await mountWithContext(
+          <Router context={{}}>
+            <Info line={externalResource} resource={externalResource} />
+          </Router>
+        );
+      });
+
+      // PARENT AGREEMENT
+      // resource = externalResource
+      it('should render parent agreement name', () => {
+        expect(interactor.parentAgreementName).to.equal(externalResource?.owner?.name);
```

Comment:
Throughout these tests we are adding the `?` check on the `externalResource` and its nested objects as its expected value. If the chaining snaps that would return the expected value to be `undefined` and you almost never want whatever you're expect()ing to be undefined. 

So if possible you could just hardcode the actual expected value in there or instead just have it as `to.equal(externalResource.owner.name)` instead?